### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by tcrossrfid**
### What changes were proposed in this pull request?

README.adoc up to date, API split off into API.adoc.

### How is this patch documented?

Documentation changes, in files mentioned above.

### How was this patch tested?

Doc changes only. Existing unit tests.

#### Depends On

Nothing.

### CODE REVIEW

File: API.adoc [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+----
+
+== Configuration
+
+Below is a typical `smartcosmos-edge-user-devkit.yml` file, which provides configuration
+for the service.  Individual endpoints can be turned off by setting their respective
+`enabled` flags to false. The default behavior (i.e., in the absence of an `enabled`
+  flag for the endpoint) is enabled.
+
+If `thread-pool.queue-capacity` is set to `-1`, the queue will have maximum capacity
```
Comment (andreassiegel) [inline API.adoc]: There is no thread pool. ;)

File: API.adoc [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+|====================
+
+.**URN Scheme**
+Note that the illustrated scheme for URNs is only for documentation purposes. There must not be any assumptions or expectations on the scheme in the REST layer. All URNs or identifiers are just `String` values in the scope of REST modules!
+
+=== Request parameters
+[width="100%",options="header",cols="1,^1,2,^1,5"]
+|====================
+| Parameter | Parameter Type |Format | Default | Description
+| `urn` | url | String | no default | the URN of the Thing
```
Comment (andreassiegel) [inline API.adoc]: URN of User Role or Tenant, not Thing.

File: API.adoc [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+| `roles` | Array of String |  | yes | e.g. `Admin` or `User`
+| `authorities` | Array of String |  | yes | e.g. `https://authorities.smartcosmos.net/things/read` or `https://authorities.smartcosmos.net/things/create`
+| `tenantUrn` | String representation of the tenant identifier in a common scheme, e.g. `urn:tenant:uuid:<UUID>` |  | yes |
+|====================
+
+=== JSON Fields for Roles
+[width="100%",options="header",cols="1,^1,2,^1,5"]
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String representation of a user identifier in a common scheme, e.g. `urn:role:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated
```
Comment (andreassiegel) [inline API.adoc]: Role identifier

File: README.adoc [hunk 4]
```diff
@@ -6,606 +6,46 @@ ifdef::env-github[:BRANCH: master]
...
-----
-[source,json]
-----
-{
-    "urn": "urn:tenant:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
-    "active": true,
-    "name": "My Example Company"
-}
-----
+* https://api.smartcosmos.net/microservices/smartcosmos-edge-user-devkit/index.html[User Management]
```
Comment (andreassiegel) [inline README.adoc]: This links to itself in the end.

Comment (tcrossrfid) [inline README.adoc]: Fixed all of the above.

**Comment by andreassiegel** [comment]
So far it looks good, but we're lacking short descriptions for the endpoints.

**Review by ghost — APPROVED** [review]

File: API.adoc [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+| `active` | boolean |  true | no |
+| `username` | String | | yes (for create) | Initial admin account
+|====================
+
+=== JSON Fields for Users
+[width="100%",options="header",cols="1,^1,2,^1,5"]
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String representation of a user identifier in a common scheme, e.g. `urn:user:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated
+| `username` | String | | yes | login name, e.g. `bob@example.com`
```
Comment (andreassiegel) [inline API.adoc]: The username doesn't need to be an email address.

File: API.adoc [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+- https://authorities.smartcosmos.net/roles/create
+- https://authorities.smartcosmos.net/roles/read
+- https://authorities.smartcosmos.net/roles/update
+- https://authorities.smartcosmos.net/roles/delete
+
+== REST API
+
+<<tenants, *Available methods for tenants*>>
+
+|====================
```
Comment (tcrossrfid) [inline API.adoc]: More consistent with everything else to leave it here at the top.

File: API.adoc [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+|<<tenantread1, Find by URN>>|`GET /tenants/{urn}`|https://authorities.smartcosmos.net/tenants/read
+|<<tenantread2, Find by Name>>|`GET /tenants/?name={name}`|https://authorities.smartcosmos.net/tenants/read
+|====================
+
+<<users, *Available methods for users*>>
+
+|====================
+|Endpoint|REST API|Required Authority
+|<<usercreate1, Create>>|`POST /users`|https://authorities.smartcosmos.net/users/create
+|<<userupdate1, Update>>|`PUT /users/{urn}`|https://authorities.smartcosmos.net/users/update
```
Comment (tcrossrfid) [inline API.adoc]: Not a doc issue :-)
Weird, though. Are you either Admin user or the user whose info is being changed?

File: API.adoc [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String representation of a tenant identifier in a common scheme, e.g. `urn:tenant:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated
+| `name` | String | | yes | Name of the tenant organization (e.g. `Example Company`)
+| `active` | boolean |  true | no |
+| `username` | String | | yes (for create) | Initial admin account
+|====================
+
+=== JSON Fields for Users
+[width="100%",options="header",cols="1,^1,2,^1,5"]
```
Comment (tcrossrfid) [inline API.adoc]: Done.

File: API.adoc [hunk 9]
```diff
@@ -0,0 +1,611 @@
...
+|====================
+| Field | Format | Default | Required | Description
+| `urn` | String representation of a user identifier in a common scheme, e.g. `urn:user:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated
```
Comment (tcrossrfid) [inline API.adoc]: Done. Also in edge-things.

File: API.adoc (hunk 5 above)
Comment (tcrossrfid) [inline API.adoc]: IMHO we should dump the email address and require the username to be parseable as an email address. But as Will points out, it's too late now.

**Pull request merged**